kolibri/auth/models.py

```diff
@@ -26,120 +38,258 @@ class KolibriValidationError(KolibriError):
     pass
 
 
-class BaseUser(AbstractBaseUser):
+##############################################
+# MORANGO: These pieces will live in Morango, but dummies/stubs are temporarily included here until Morango is ready.
+##############################################
```

Comment:
That said, regarding the justification you give for `FacilityDataset` I notice that two of the reasons have nothing to do with Morango, and one does. I caution against creating a class that has unrelated responsibilities. There *is* a concrete justification for this class arising purely from within this module, as you point out -- what I'm suggesting is to stick close to those reasons.

For instance, does it make sense for a `FacilityDataset` to exist without a `Collection` of kind "facility"? Should `FacilityDataset` be the "privileged" model, or should `Collection`s with kind "facility"? By "privileged" I mean should it be the foreign-key for objects that are namespaced by facilities. Sadly, without a more in-depth consideration that we don't seem to have the time for, I can only appeal to my intuition. (My intuition answers "no" and "`Collection`s", FWIW.)

As far as the implementation, what you're describing sounds very much like a [one-to-one relationship](https://docs.djangoproject.com/en/1.9/topics/db/examples/one_to_one/), so you might consider if that's appropriate here.